### PR TITLE
remove our WebSocket client

### LINKED ISSUE #108: Remove our WebSocket Client
CoreFX has a client that works on Win8+ and there are discussions about making it work cross-plat. Our client was Desktop-only anyway since it depended on special features in HttpWebRequest so it only ran on Windows or in Mono.

### PR DESCRIPTION
**Pull request by anurse**
we'll use the one from dotnet/corefx now instead

fixes #108

/cc @Tratcher

### CODE REVIEW

File: test/Microsoft.AspNetCore.WebSockets.Test/WebSocketMiddlewareTests.cs [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+
+using System;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Testing.xunit;
+using Xunit;
+
+namespace Microsoft.AspNetCore.WebSockets.Test
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.WebSockets.Test/WebSocketMiddlewareTests.cs]: We usually put the functional tests in a .FunctionalTests project, separate from the unit tests.

Comment (Tratcher) [inline test/Microsoft.AspNetCore.WebSockets.Test/WebSocketMiddlewareTests.cs]: The best boundary I've seen is if we had to open a Socket and spin up a full host.

**Comment by Tratcher** [comment]
:shipit:

File: test/Microsoft.AspNetCore.WebSockets.Test/WebSocketMiddlewareTests.cs (hunk 1 above)
Comment (anurse) [inline test/Microsoft.AspNetCore.WebSockets.Test/WebSocketMiddlewareTests.cs]: Yeah, I can separate them. I was just copying the existing tests around and they hadn't been separated so I didn't do a lot of looking at the layout :)

File: test/Microsoft.AspNetCore.WebSockets.Test/project.json [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+    "Microsoft.AspNetCore.WebSockets": "0.2.0-*",
+    "Microsoft.AspNetCore.Server.Kestrel": "1.1.0-*",
     "xunit": "2.2.0-*"
   },
   "testRunner": "xunit",
   "frameworks": {
+    "net451": {},
     "netcoreapp1.0": {
       "dependencies": {
+        "System.Net.Websockets.Client": "4.0.0",
```
Comment (agoretsky) [inline test/Microsoft.AspNetCore.WebSockets.Test/project.json]: System.Net.Websockets.Client should be System.Net.WebSockets.Client

**Pull request merged**